basis/editors/brackets/brackets.factor

```diff
@@ -0,0 +1,33 @@
+! Copyright (C) 2015 Dimage Sapelkin.
+! See http://factorcode.org/license.txt for BSD license.
+USING: editors kernel make math.parser namespaces sequences quotations system alien.data alien.strings 
+       windows.advapi32 windows.registry windows.types windows.errors windows.registry.private combinators ;
+IN: editors.brackets
+
+SINGLETON: brackets-editor
+brackets-editor \ editor-class set-global
+
+! SYMBOL: brackets-path
+
+! HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\Brackets.exe
+
+<PRIVATE
+
+: windows-get-brackets-path ( -- path )
```

Comment:
Another way to do the ``os`` dispatch is:

```factor
HOOK: brackets-path os ( -- path )

M: windows brackets-path
    HKEY_LOCAL_MACHINE ...
```

Then in ``M\ brackets-editor editor-command`` you can just do:

```factor
M: brackets-editor editor-command
    [ brackets-path "brackets" or , drop , ] { } make ;
```